docs/v1/api.md

```diff
@@ -0,0 +1,239 @@
+
+## Persistent Volumes
+
+First, this problem space is similar to that of persistent volumes. We model the data structures herein after those in persistent volumes primarily because we believe that consistency is a virtue. More importantly, the community benefits when similar problems have similar solutions.
+
+Specifically, we believe that the following existing types provide a good model for the mechanics of how the services offered by a broker can be used in Kubernetes:
+
+- `StorageClass`
+- `PersistentVolume`
+- `PersistentVolumeClaim`
+
+## Ingress Controllers
+
+Today, operators choose their own implementation of an ingress controller. Kubernetes gives the operator the ability to specify any ingress controller, and the complete freedom to implement it themselves.
+
+Similarly, permitting operators to implement or select their own service catalog controller would encourage freedom of choice and diversity. The analog to an ingress controller for the service catalog system is called a _service catalog controller_. Like ingress, these controllers run inside a standard pod. Anyone can write and deploy a service catalog controller, with no changes necessary to the Kubernetes install.
+
+Service catalog controllers vary, however, from ingress controllers, because a cluster can consist of many different controllers, each of which handles actions on a subset of the available service-related resources. For example, a cluster can run a service controller to handle AWS RDS databases, and a completely separate one to handle AWS Elastic Load Balancer instances.
+
+As we'll detail below, each service controller implementation must to conform to an implementation spec, but that spec allows for a heterogenous set of controllers in a cluster. We believe this allowance encourages flexibility in a cluster's service catalog.
+
+# Kubernetes Resources
+
+Given these inspirations, the following resources will be added to the Kubernetes API:
+
+- `ServiceClass`
+  - Describes a _kind_ of service. This resource is semantically similar to how a `StorageClass` defines a _kind_ of persistent volume, but we expect that, in practice, the kinds of services represented here will be more coarse grained than the kinds of volumes represented in `StorageClass`es
+  - References a service ID and plan ID, which is managed and understood by a service controller (see below)
+  - Created by an operator or tool
+- `ServiceInstance`
+  - An instance of a `ServiceClass` -- an in-cluster representation of a provisioned service
+  - Can be statically provisioned by an operator, to enable multiple consumers to bind to it (enables multi-tenancy)
+  - Can be dynamically provisioned and bound to by a consumer, to guarantee an app’s exclusive use of the resource (guarantees single-tenancy)
+- `ServiceInstanceClaim`
+  - Created by a consumer to bind to an existing ServiceInstance by name. As written above, this workflow enables multi-tenant usage of a resource
```

Comment:
I was going to agree with this too :-)  but then I think I had an ah-ha moment.  Let's see how far off I am....
the data in a ServiceBinding is generated by a service broker, not by an end-user.  So, in order to trigger the service controller to initiate the "CreateServiceBinding()" call to the SB we need something for the controler to notice - hence the claim.  This all comes back to the fundamental idea that Kube is nothing more than an API wrapper for a DB - there really is no notion of the end-user asking for something to happen - directly. Its all indirectly done via REST/CRUD calls on resources.

Am I even close? 

If I'm right, I'm not happy about it, but I think this is "the k8s way".  Are there any examples in k8s that don't follow this model?